contracts/vault/AssetVault.sol

```diff
@@ -173,6 +174,24 @@ contract AssetVault is IAssetVault, OwnableERC721, Initializable, ERC1155Holder,
         emit WithdrawETH(msg.sender, to, balance);
     }
 
+    /**
+     * @notice Withdraw cryptoPunk from the vault.
+     *         The vault must be in a "withdrawEnabled" state (non-transferrable),
+     *         and the caller must be the owner.
+     *
+     * @param punks                 The CryptoPunk contract address.
+     * @param punkIndex             The index of the CryptoPunk to withdraw (i.e. token ID).
+     * @param to                    The recipient of the withdrawn punk.
+     */
+    function withdrawPunk(
+        address punks,
+        uint256 punkIndex,
+        address to
+    ) external onlyOwner onlyWithdrawEnabled {
+        IPunks(punks).transferPunk(to, punkIndex);
+        emit WithdrawERC721(msg.sender, punks, to, punkIndex);
```

Comment:
Since it's not "technically" an `ERC721`, we should probably define a new `WithdrawPunk` event. It can have the same four arguments, just a different name. Otherwise I think it might confuse the subgraph.